akka-http/src/main/scala/akka/http/scaladsl/common/JsonSourceRenderingMode.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.common
+
+import akka.http.scaladsl.model.{ ContentType, ContentTypes }
+import akka.util.ByteString
+
+/**
+ * Specialised rendering mode for streaming elements as JSON.
+ *
+ * See also: <a href="https://en.wikipedia.org/wiki/JSON_Streaming">JSON Streaming on Wikipedia</a>.
+ *
+ * See [[JsonSourceRenderingModes]] for commonly used pre-defined rendering modes.
+ */
+trait JsonSourceRenderingMode extends akka.http.javadsl.common.JsonSourceRenderingMode with SourceRenderingMode {
+  override val contentType: ContentType.WithFixedCharset =
+    ContentTypes.`application/json`
+}
+
+/**
+ * Provides default JSON rendering modes.
+ */
+object JsonSourceRenderingModes {
```

Comment:
Well, `LineByLine` would just be a `JsonSourceRenderingMode()` (the default) while the other would be
```scala
implict val renderingMode = JsonSourceRenderingMode("[", ",", "]")
```

Not sure that's worth the extra object. Maybe the compact array above could be included as an example in the (Scala-)Doc.